src/plugins/discover/public/application/context/services/context_state.test.ts

```diff
@@ -169,7 +169,7 @@ describe('Test Discover Context State', () => {
     `);
     state.flushToUrl();
     expect(getCurrentUrl()).toMatchInlineSnapshot(
-      `"/#?_g=(filters:!(('$state':(store:globalState),meta:(alias:!n,disabled:!f,index:'logstash-*',key:extension,negate:!f,params:(query:jpg),type:phrase),query:(match_phrase:(extension:(query:jpg))))))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'logstash-*',key:extension,negate:!t,params:(query:png),type:phrase),query:(match_phrase:(extension:(query:png))))),predecessorCount:4,successorCount:4)"`
+      `"/#?_g=(filters:!(('$state':(store:globalState),meta:(alias:!n,disabled:!f,index:'logstash-*',key:extension,negate:!f,params:(query:jpg),type:phrase,value:jpg),query:(match_phrase:(extension:(query:jpg))))))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'logstash-*',key:extension,negate:!t,params:(query:png),type:phrase,value:png),query:(match_phrase:(extension:(query:png))))),predecessorCount:4,successorCount:4)"`
```

Comment:
I've question, I noticed in this PR that now the value field is also part of the URL. My question: since this wasn't here before, is this needed? we have the same value in param.query. also then in the match_phrase query. Or wouldn't it be better to just use 'value:query' in state and generate the query by this value?

Reply:
Yes, this does seem completely redundant, but the type definition for filters specifies that this must be included: https://github.com/elastic/kibana/blob/a296e4cc970033c1a466b60b46b16136d9a35699/packages/kbn-es-query/src/filters/build_filters/types.ts#L67

In the future we will want to remove the `query` portion of our filter definitions (except for custom filters), so it won't longer be redundant. But we're still a way off from that.

Reply:
Well, after I wrote this comment I realized it's not actually "required" since it's optional. I've reverted this test to not include it in the URL.